chromadb/auth/basic_authn/__init__.py

```diff
@@ -100,11 +102,11 @@ def __init__(self, system: System) -> None:
         "BasicAuthenticationServerProvider.authenticate", OpenTelemetryGranularity.ALL
     )
     @override
-    def authenticate_or_raise(self, headers: Headers) -> UserIdentity:
+    def authenticate_or_raise(self, headers: Dict[str, str]) -> UserIdentity:
```

Comment:
This is a breaking change for anyone overriding auth. We should be more careful here.

Reply:
Agreed. But the use of `starlette` and `fastapi` here was already a breaking change for the thin client, which did not require (or import) either of those.  Technically, duck-typing helps us here too.